### PR TITLE
Report whether the pattern miner is compiled

### PR DESCRIPTION
**Pull request by ngeiswei**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -694,4 +694,5 @@ SUMMARY_ADD("Server" "The CogServer" HAVE_SERVER)
 SUMMARY_ADD("Spatial" "Vizualisation of the 3D spacemap" HAVE_SPATIAL_TOOLS AND HAVE_OCTOMAP)
 SUMMARY_ADD("Unit tests" "Unit tests" CXXTEST_FOUND)
 SUMMARY_ADD("NLP" "Natural Language Processing" HAVE_NLP)
+SUMMARY_ADD("PatternMiner" "Pattern Miner" HAVE_cpprest)
```
Comment (linas) [inline CMakeLists.txt]: why in the world would the pattern miner depend on the REST interfaces??

Comment (linas) [inline CMakeLists.txt]: Oh, never mind, I see it has a client-server mode.

Comment (ngeiswei) [inline CMakeLists.txt]: Perhaps what I should have done is define a cmake constant HAVE_PATTERN_MINER and use that instead, but I figured we can probably do that later on if the pattern miner requirements evolves.

Comment (linas) [inline CMakeLists.txt]: See #2225 its low priority, though

**Pull request merged**